spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/services/SCBackendService.java

```diff
@@ -548,107 +548,109 @@ private String buildDeviceInfo(String token) {
     }
 
     private void listenForSyncEvents() {
-
         Observable<SCDataStore> syncableStores = dataService.getISyncableStores();
 
-        Observable<Boolean>  storeEditSync = syncableStores
-                .flatMap(new Func1<SCDataStore, Observable<SCSpatialFeature>>() {
+        Observable<Observable> storeEditSync =
+                syncableStores.flatMap(new Func1<SCDataStore, Observable<SCSpatialFeature>>() {
                     @Override
                     public Observable<SCSpatialFeature> call(SCDataStore scDataStore) {
                         return scDataStore.storeEdited;
                     }
-                }).map(new Func1<SCSpatialFeature, Boolean>() {
+                }).map(new Func1<SCSpatialFeature, Observable>() {
                     @Override
-                    public Boolean call(SCSpatialFeature feature) {
-                        final ISyncableStore store = (ISyncableStore) dataService.getStoreByIdentifier(feature.getStoreId());
-                        syncStore(store);
-                        return null;
+                    public Observable call(SCSpatialFeature feature) {
+                        final ISyncableStore store = (ISyncableStore) dataService
+                                .getStoreByIdentifier(feature.getStoreId());
+                        return syncStore(store);
                     }
                 });
 
         //sync all stores when connection to broker is true
-        Observable<Boolean> onlineSync = connectedToBroker
-                .filter(new Func1<Boolean, Boolean>() {
-                    @Override
-                    public Boolean call(Boolean sync) {
-                        return sync;
-                    }
-                }).map(new Func1<Boolean, Boolean>() {
-                    @Override
-                    public Boolean call(Boolean sync) {
-                        syncStores();
-                        return sync;
-                    }
-                });
+        Observable<Observable> onlineSync = connectedToBroker.filter(new Func1<Boolean, Boolean>() {
+            @Override
+            public Boolean call(Boolean sync) {
+                return sync;
+            }
+            // for each connected State true (sync is true), sync the stores
+        }).map(new Func1<Boolean, Observable>() {
+            @Override
+            public Observable call(Boolean sync) {
+                return syncStores();
+            }
+        });
 
-        Observable<Boolean> sync = onlineSync.mergeWith(storeEditSync);
+        Observable<Observable> sync = onlineSync.mergeWith(storeEditSync);
         sync.subscribeOn(Schedulers.newThread()).subscribe();
```

Comment:
probably need to catch the onError here,  like iOS

https://github.com/boundlessgeo/spatialconnect-ios-sdk/blob/master/SpatialConnect/SCBackendService.m#L326-L332